src/Coordination/Changelog.cpp

```diff
@@ -540,23 +546,62 @@ ChangelogRecord Changelog::buildRecord(uint64_t index, const LogEntryPtr & log_e
     return record;
 }
 
+void Changelog::writeThread()
+{
+    // we don't protect current_writer because rotate at the same time can be called from compact only
+    // when the node is applying snapshot from leader
+    // in that case, no entry write should happen, i.e. this thread will be inactive
+    WriteOperation write_operation;
+    while (write_operations.pop(write_operation))
+    {
+        std::visit([&, this]<typename WriteOperationType>(const WriteOperationType & operation) -> void
+        {
+            if constexpr (std::same_as<WriteOperationType, AppendLog>)
```

Comment:
To me, the code without `std::visit` and `if constexpr` would be simpler to read.